simphony/io/h5_cuds.py

```diff
@@ -152,6 +155,47 @@ def add_mesh(self, mesh):
         self._file.flush()
         return m
 
+    def add_lattice(self, lattice, record=None):
```

Comment:
> and a 19-dimensional 64bit float vector, that is used internally in the simulation engine.

Is this related to a CUBA keywords or is part of the internal representation of a specific engine? Do all engines will need to save these values? If the float vector was part of the CUBA keywords that are saved wouldn't that remove the need for extended the table columns?

>  I hope that we can somehow allow partial adding of the CUBA-keys with add_lattice so that it would still work similarly to the other CUDS-types and their related H5CUDS-methods. 

If such a requirement is necessary, then an `extend_lattice` method would be more appropriate.

> I also think that partial adding, in the case of Lattice-Boltzmann, will be the more often used scenario.

If Table extension is taking place all the time then why do we use a restricted record?



I would suggest that the `records` attribute is not part of this PR and a SSB discussion is initiated (similar to what is taking place now with sharing) regarding the usecase and possible solutions to the issues that are raised. @nathanfranklin, @kemattil, @ahashibon what do you think.